### PR TITLE
feat(html): Add BreakdownTable.ts

### PR DESCRIPTION
**Pull request by jbrundage**
## Checklist:
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit message includes a "fixes" reference if appropriate.
  - [x] The commit is signed.
- [x] The change has been fully tested:
  - [x] I have viewed all related gallery items
  - [ ] I have viewed all related dermatology items
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised new issues to address them separately

## Testing:
Tested gallery sample

### CODE REVIEW

**Comment by jbrundage** [comment]
image

File: packages/html/src/BreakdownTable.ts [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+            ;
+    }
+    update(domNode, element) {
+        const rowCount = this.useCalculatedRowCount() ? this.calculateRowCount() : this.rowCount();
+        this.transformData = function() {
+            return this.breakdownData(rowCount);
+        };
+
+        this._tooltip
+            .tooltipHTML(data => {
```
Comment (GordonSmith) [inline packages/html/src/BreakdownTable.ts]: Does this function have to be reset every update?

File: packages/html/src/BreakdownTable.ts [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+                        .direction("n")
+                        .data(context.data())
+                        .visible(true)
+                        .render()
+                        ;
+                })
+                ;
+        }
+    }
+    breakdownData(limit: number) {
```
Comment (GordonSmith) [inline packages/html/src/BreakdownTable.ts]: Mark as `protected` - move to above enter

File: packages/html/src/BreakdownTable.ts [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+                data.push([row[0], perc + "%"]);
+            });
+        if (showOther) {
+            const otherLabel = `${this.otherLabel()} (${len - limit + 1})`;
+            const otherPercentage = "~" + (100 - percSum) + "%";
+            data.push([otherLabel, otherPercentage]);
+        }
+        return data;
+    }
+    calculateRowCount() {
```
Comment (GordonSmith) [inline packages/html/src/BreakdownTable.ts]: Mark as protected and move to before enter.

File: packages/html/src/BreakdownTable.ts [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+        const theadRowHeight = this.columns().length > 0 ? this.thFontSize() + 5 : 0;
+        const tbodyRowHeight = this.fontSize() + 5;
+        const tbodyAvailableHeight = this.height() - theadRowHeight;
+        const rowCount = Math.floor(tbodyAvailableHeight / tbodyRowHeight);
```
Comment (GordonSmith) [inline packages/html/src/BreakdownTable.ts]: No need for `rowCount`

File: packages/html/src/BreakdownTable.ts [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+        return rowCount;
+    }
+    tooltip;
+    tooltipHTML: (_) => string;
+    tooltipFormat: (_) => string;
```
Comment (GordonSmith) [inline packages/html/src/BreakdownTable.ts]: What are these three items for?

File: packages/html/src/BreakdownTable.ts (hunk 1 above)
Comment (jbrundage) [reply packages/html/src/BreakdownTable.ts]: It's reset on every update because the data and/or parameters may change.

Comment (GordonSmith) [reply packages/html/src/BreakdownTable.ts]: But its not accessing the data at this point in time (or any of the values as far as I can tell)?

Comment (GordonSmith) [reply packages/html/src/BreakdownTable.ts]: (It is using RowCount, but that is the only local value, which could be moved into the function?

Comment (jbrundage) [reply packages/html/src/BreakdownTable.ts]: Valid, I'll move this to enter.

**Pull request closed without merging**